website/server/controllers/api-v3/members.js

```diff
@@ -57,47 +57,81 @@ api.getMember = {
   },
 };
 
-function _addAchievement(achievementsResult, type, key, title, text, icon, category, value, earned) {
-  achievementsResult[key] = {
-    type,
-    title,
-    text,
+
+function _addAchievement (achievementsResult, data) {
+  achievementsResult[data.key] = {
+    type: data.type,
+    title: data.title,
+    text: data.text,
+    icon: data.icon,
+    category: data.category,
+    key: data.key,
+    value: data.value,
+    earned: data.earned,
+    index: achievementsResult._count++,
+  };
+}
+
+function _addQuestAchievement (res, memberObj, achievementsResult, key, icon, category, labels) {
```

Comment:
`res`, `memberObj` and `achievementsResult` also inside the `data` object? I'd rather have those as separate params. @khipkin / @crookedneighbor  your opinion please ^^